src/main/java/core/basesyntax/service/impl/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitStorageDao storageDao;
+
+    public PurchaseOperation(FruitStorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void doOperation(String fruitName, Integer quantity) {
+        if (quantity < 0) {
+            throw new IllegalArgumentException(
+                    "Quantity cannot be negative for purchase operation: " + quantity
+            );
+        }
```

Comment:
You can move it to the `default` (maybe better choice) or `static` method in the OperationHandler interface to avoid code duplication